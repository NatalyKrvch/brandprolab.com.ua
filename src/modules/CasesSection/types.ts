import { CasesSectionType } from '@/sanity/lib/types';

export interface CasesSectionProps {
  casesData: CasesSectionType;
}
