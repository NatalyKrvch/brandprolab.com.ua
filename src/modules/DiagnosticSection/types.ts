import { DiagnosticSectionType } from '@/sanity/lib/types';

export interface DiagnosticSectionProps {
  diagnosticData: DiagnosticSectionType;
}
