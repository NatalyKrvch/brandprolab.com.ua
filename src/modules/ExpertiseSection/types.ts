import { ExpertiseSectionType } from '@/sanity/lib/types';

export interface ExpertiseSectionProps {
  expertiseData: ExpertiseSectionType;
}
