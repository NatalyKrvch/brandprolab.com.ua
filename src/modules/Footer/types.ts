import { FooterSectionType } from '@/sanity/lib/types';

export interface FooterProps {
  footerData: FooterSectionType;
}
