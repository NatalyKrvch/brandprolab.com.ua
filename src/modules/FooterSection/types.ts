export interface FooterSectionProps {
  offerURL: string;
}
