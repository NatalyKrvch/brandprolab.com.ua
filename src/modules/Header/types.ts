import { HeaderSectionType } from '@/sanity/lib/types';

export interface HeaderProps {
  headerData: HeaderSectionType;
  id: string;
}
