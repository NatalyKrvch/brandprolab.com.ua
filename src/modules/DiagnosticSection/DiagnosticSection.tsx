import {
  Button,
  ExternalLink,
  List,
  SimpleCard,
  SprayBackground,
} from '@/components';
import { normalizeText } from '@/utils';

import type { DiagnosticSectionProps } from './types';

const DiagnosticSection = ({ diagnosticData }: DiagnosticSectionProps) => {
  const {
    title,
    subtitle,
    description,
    benefitsTitle,
    benefits,
    buttonText,
    buttonLink,
  } = diagnosticData;

  return (
    <div className="relative w-full desktop:px-72">
      <div className="grid desktop:grid-cols-2 desktop:gap-24">
        <div className="flex flex-col items-center text-center desktop:w-464 desktop:items-start desktop:px-2 desktop:text-left">
          <h2 className="mb-2 text-4xl font-bold leading-9 tracking-tight text-black tablet:mb-3 tablet:text-44 tablet:leading-44 desktop:mb-5 desktop:text-52 desktop:leading-54">
            {normalizeText(title)}
          </h2>

          <p className="mb-8 w-190 text-xl font-normal leading-5 tracking-normal text-blue-default tablet:mb-9 tablet:w-400 tablet:px-16 tablet:text-left tablet:text-28 tablet:leading-7 desktop:mb-10 desktop:w-442 desktop:px-16 desktop:text-32 desktop:leading-8">
            {normalizeText(subtitle)}
          </p>

          <p className="mb-8 w-full max-w-400 text-base font-normal leading-5 tracking-normal text-gray-dark tablet:mb-5 tablet:max-w-482 tablet:text-lg tablet:leading-6 desktop:mb-12 desktop:max-w-full desktop:leading-6">
            {normalizeText(description)}
          </p>

          <ExternalLink
            aria-label="Перейти до запису на консультацію"
            className="hidden desktop:block desktop:w-400"
            href={buttonLink.trim()}
          >
            <Button>{normalizeText(buttonText)}</Button>
          </ExternalLink>
        </div>

        <div className="relative flex items-center justify-center">
          <SimpleCard
            backgroundColor="bg-teal-light"
            className="w-full max-w-400 p-6 tablet:max-w-482 desktop:max-w-408"
          >
            <SprayBackground className="-left-20 -top-250 h-700 w-full -rotate-30 tablet:-left-30 tablet:-top-336 tablet:h-900 tablet:w-full desktop:-top-270 desktop:h-800 desktop:w-800" />

            <p className="mb-6 font-normal text-black tablet:text-2xl tablet:leading-6 desktop:text-2xl desktop:leading-6">
              {normalizeText(benefitsTitle)}
            </p>

            <List list={benefits} />
          </SimpleCard>
        </div>

        <div className="mt-4 flex justify-center desktop:hidden">
          <ExternalLink
            aria-label="Перейти до запису на консультацію"
            className="w-full max-w-400 tablet:max-w-482 desktop:max-w-314"
            href={buttonLink.trim()}
          >
            <Button>{normalizeText(buttonText)}</Button>
          </ExternalLink>
        </div>
      </div>
    </div>
  );
};

export default DiagnosticSection;
