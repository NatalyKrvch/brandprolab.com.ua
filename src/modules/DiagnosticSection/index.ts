export { default as DiagnosticSection } from './DiagnosticSection';
export * from './types';
