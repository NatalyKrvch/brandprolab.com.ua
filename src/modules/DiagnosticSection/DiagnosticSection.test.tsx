import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import DiagnosticSection from './DiagnosticSection';
import { DiagnosticSectionProps } from './types';

jest.mock('@/sanity/lib/client', () => ({
  client: {
    fetch: jest.fn(),
  },
}));

jest.mock('@/components', () => ({
  Button: ({ children }: { children: React.ReactNode }) => (
    <button>{children}</button>
  ),
  ExternalLink: ({ children }: { children: React.ReactNode }) => (
    <a>{children}</a>
  ),
  List: ({ list }: { list: string[] }) => (
    <ul>
      {list.map((item, index) => (
        <li key={index}>{item}</li>
      ))}
    </ul>
  ),
  SimpleCard: ({ children }: { children: React.ReactNode }) => (
    <div>{children}</div>
  ),
  SprayBackground: () => <div>SprayBackground</div>,
}));

describe('DiagnosticSection component', () => {
  const mockProps: DiagnosticSectionProps = {
    diagnosticData: {
      _id: 'test-id',
      _type: 'diagnosticSection',
      title: 'Some title',
      subtitle: 'Some subtitle',
      description: 'Some description',
      benefitsTitle: 'Benefits title',
      benefits: ['Benefit 1', 'Benefit 2'],
      buttonText: 'Click Me',
      buttonLink: '/offer',
    },
  };

  it('renders DiagnosticSection correctly', () => {
    const { asFragment } = render(<DiagnosticSection {...mockProps} />);

    expect(screen.getByText('Some title')).toBeInTheDocument();
    expect(screen.getByText('Some subtitle')).toBeInTheDocument();
    expect(screen.getByText('Some description')).toBeInTheDocument();
    expect(screen.getByText('Benefits title')).toBeInTheDocument();
    expect(screen.getByText('Benefit 1')).toBeInTheDocument();
    expect(screen.getByText('Benefit 2')).toBeInTheDocument();
    expect(screen.getAllByText('Click Me')).toHaveLength(2);

    expect(asFragment()).toMatchSnapshot();
  });
});
