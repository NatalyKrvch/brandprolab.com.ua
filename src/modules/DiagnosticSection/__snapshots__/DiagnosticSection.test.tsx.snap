// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiagnosticSection component renders DiagnosticSection correctly 1`] = `
<DocumentFragment>
  <section
    class="relative w-full desktop:px-72"
  >
    <div
      class="grid desktop:grid-cols-2 desktop:gap-24"
    >
      <div
        class="flex flex-col items-center text-center desktop:w-464 desktop:items-start desktop:px-2 desktop:text-left"
      >
        <h2
          class="mb-2 text-4xl font-bold leading-9 tracking-tight text-black tablet:mb-3 tablet:text-44 tablet:leading-44 desktop:mb-5 desktop:text-52 desktop:leading-54"
        >
          Some title
        </h2>
        <p
          class="mb-8 w-190 text-xl font-normal leading-5 tracking-normal text-blue-default tablet:mb-9 tablet:w-400 tablet:px-16 tablet:text-left tablet:text-28 tablet:leading-7 desktop:mb-10 desktop:w-442 desktop:px-16 desktop:text-32 desktop:leading-8"
        >
          Some subtitle
        </p>
        <p
          class="mb-8 w-full max-w-400 text-base font-normal leading-5 tracking-normal text-gray-dark tablet:mb-5 tablet:max-w-482 tablet:text-lg tablet:leading-6 desktop:mb-12 desktop:max-w-full desktop:leading-6"
        >
          Some description
        </p>
        <a>
          <button>
            Click Me
          </button>
        </a>
      </div>
      <div
        class="relative flex items-center justify-center"
      >
        <div>
          <div>
            SprayBackground
          </div>
          <p
            class="mb-6 font-normal text-black tablet:text-2xl tablet:leading-6 desktop:text-2xl desktop:leading-6"
          >
            Benefits title
          </p>
          <ul>
            <li>
              Benefit 1
            </li>
            <li>
              Benefit 2
            </li>
          </ul>
        </div>
      </div>
      <div
        class="mt-4 flex justify-center desktop:hidden"
      >
        <a>
          <button>
            Click Me
          </button>
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
